RadixWallet/Features/DappInteractionFeature/Coordinator/DappInteractionLoading.swift

```diff
@@ -106,7 +114,60 @@ struct DappInteractionLoading: Sendable, FeatureReducer {
 						return .failure(error)
 					}
 					loggerGlobal.warning("Failed to fetch Dapps metadata, but since 'isDeveloperModeEnabled' is enabled we surpress the error and allow continuation. Error: \(error)")
-					return .success(.request(request))
+					return .success(.request(metadata))
+				}
+
+			}()
+
+			await send(.internal(.dappMetadataLoadingResult(result)))
+		}
+	}
+
+	private func deeplinkMetadataEffect(items: P2P.Dapp.Request.VerifyItems, metadata: P2P.Dapp.Request.Metadata) -> Effect<Action> {
+		.run { send in
+			guard
+				let wellKnownFile = try? await httpClient.fetchDappWellKnownFile(items.dappOrigin),
+				let dappDefinitionAddress = wellKnownFile.dApps.first?.dAppDefinitionAddress
+			else {
+				struct MissingDappDefinitionAddress: Error {}
+				throw MissingDappDefinitionAddress()
+			}
+
+			let result: TaskResult<DappMetadata> = await {
+				let isDeveloperModeEnabled = await appPreferencesClient.getPreferences().security.isDeveloperModeEnabled
+				do {
+					let cachedMetadata = try await cacheClient.withCaching(
+						cacheEntry: .dAppRequestMetadata(dappDefinitionAddress.address),
+						invalidateCached: { (cached: DappMetadata.DeepLink) in
+							guard
+								cached.name != nil,
+								cached.description != nil,
+								cached.thumbnail != nil
+							else {
+								/// Some of these fields were not set, fetch and see if they
+								/// have been updated since last time...
+								return .cachedIsInvalid
+							}
+							// All relevant fields are set, the cached metadata is valid.
+							return .cachedIsValid
+						},
+						request: {
+							let entityMetadataForDapp = try await gatewayAPIClient.getEntityMetadata(dappDefinitionAddress.address, .dappMetadataKeys)
+							return .init(
+								entityMetadataForDapp: entityMetadataForDapp,
+								dAppDefinintionAddress: dappDefinitionAddress,
+								origin: items.dappOrigin,
+								wellKnownFile: wellKnownFile
```

Comment:
We need to keep the wellKnownFile, and that's basically why we use `.deeplink` and not `.ledger` (although I imagine you thought of it this way as well when adding the `case deeplink`)